app/src/androidTest/java/br/com/simplepass/loadingbutton/Wait.kt

```diff
@@ -0,0 +1,81 @@
+package br.com.simplepass.loadingbutton
+
+import android.view.View
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.FailureHandler
+import androidx.test.espresso.Root
+import androidx.test.espresso.ViewAssertion
+import androidx.test.espresso.matcher.RootMatchers
+import org.hamcrest.Matcher
+
+class Wait private constructor(
+    private val delay: Long,
+    private val duration: Long,
+    private val startTime: Long,
+    private val viewMatcher: Matcher<View>,
+    private val rootMatcher: Matcher<Root>,
+    private val assertion: ViewAssertion
+) : FailureHandler {
+
+    init {
+        checkAssertion()
+    }
+
+    override fun handle(error: Throwable?, viewMatcher: Matcher<View>?) {
+        val currentTime = System.currentTimeMillis()
+
+        if (currentTime > (startTime + duration)) {
+            throw WaitingTimeoutException(error)
+        }
+
+        try {
+            Thread.sleep(delay)
+        } catch (e: InterruptedException) {
+            // let it go
+        }
+
+        checkAssertion()
+    }
+
+    private fun checkAssertion() {
+        Espresso.onView(viewMatcher)
+                .inRoot(rootMatcher)
+                .withFailureHandler(this)
+                .check(assertion)
+    }
+
+    class Builder constructor(private val viewMatcher: Matcher<View>) {
+        private var delay: Long = 50
+        private var duration: Long = 5000
+        private var rootMatcher: Matcher<Root> = RootMatchers.DEFAULT
+
+        fun inRoot(matcher: Matcher<Root>): Builder {
```

Comment:
`apply` could be used here to shrink the code. It can be used in the whole builder instead of returning `this`

Reply:
example:

```
fun duration(duration: Long): Builder = apply {
            this.duration = duration
        }
```